src/app/pages/storage2/modules/devices/components/devices/devices.component.ts

```diff
@@ -73,39 +77,42 @@ export class DevicesComponent implements OnInit {
     this.dataSource = new IxNestedTreeDataSource(dataNodes);
     this.dataSource.filterPredicate = (dataNodes, query = '') => {
       return flattenTreeWithFilter(dataNodes, (dataNode) => {
-        if (isVDev(dataNode)) {
-          switch (dataNode.type) {
-            case VDevType.Disk:
-              return dataNode.disk?.toLowerCase().includes(query.toLowerCase());
-            case VDevType.Mirror:
-              return dataNode.name?.toLowerCase().includes(query.toLowerCase());
-          }
-        } else {
+        if (isVdevGroup(dataNode)) {
```

Comment:
Previously we were checking for `'stat' in obj` for this check which we're not doing in this code block now. Just want to make sure we're changing the outcome of this conditional block.

Reply:
The outcome is the same, changes names of functions and uses return early.